### ISSUE TITLE
Graph with allows_loops = False, has_loops = False and num_edges = 0 returns loop in edge_iterator

### ISSUE THREAD

**Issue by cyrilbouvier** [sagemath/sage#34939]
<div id="comment:0"></div>

Consider the following code:

```
sage: G = Graph()
sage: G.add_vertex()
0
sage: G.allow_loops(True)
sage: G.add_edge (0, 0)
sage: G.allow_loops(False, check=False)
sage: G.allows_loops()
False
sage: G.has_loops()
False
sage: G.num_edges()
0
sage: list(G.edge_iterator())
[(0, 0, None)]
```

The root of the problem is that using `check=False` in `allow_loops (False, check=False)` does not remove the existing loop(s). I know that it is the goal of this keyword but it should, in case of existing loops, raise an exception (like when one tries to add a loop on a graph where loops are not allowed)

By the way, in my opinion, the name `check` for the keyword argument of `allow_loops` should be rename to something like `delete_existing_loops`.

Edit: there is a similar situation with multiple edges:

```
sage: G = Graph()
sage: G.add_vertices ([0, 1])
sage: G.add_edge (0, 1)
sage: G.add_edge (0, 1)
sage: G.num_edges()
1
sage: list(G.edge_iterator())
[(0, 1, None)]
sage: G.allow_multiple_edges(True)
sage: G.add_edge (0, 1, 'label')
sage: G.num_edges()
2
sage: list(G.edge_iterator())
[(0, 1, None), (0, 1, 'label')]
sage: 
sage: G.allow_multiple_edges (False, check=False)
sage: G.allows_multiple_edges()
False
sage: G.has_multiple_edges()
False
sage: G.num_edges()
2
sage: list(G.edge_iterator()) # only one edge but num_edges = 2 !!
[(0, 1, None)]
sage: 
sage: G.allow_multiple_edges(True)
sage: G.num_edges()
2
sage: list(G.edge_iterator())
[(0, 1, None), (0, 1, 'label')]
```

Component: **graph theory**

_Issue created by migration from https://trac.sagemath.org/ticket/34939_

**Comment by aniketwdubey** [comment]
You are correct that using check=False in allow_loops(False, check=False) does not remove existing loops, but rather just changes the property of the graph to disallow new loops. This behavior is because allow_loops is primarily intended to change the graph's properties, rather than modify its structure.

However, you make a valid point that this behavior can be confusing and could potentially lead to unexpected results. A more precise and explicit way to remove existing loops would be to delete them using the delete_edge method manually.

In this particular case, if you wanted to disallow loops and remove any existing loops, you could modify the code as follows:

`sage: G = Graph()
sage: G.add_vertex()
0
sage: G.allow_loops(True)
sage: G.add_edge(0, 0)
sage: G.delete_edge(0, 0)  # manually remove the existing loop
sage: G.allow_loops(False)
sage: G.allows_loops()
False
sage: G.has_loops()
False
sage: G.num_edges()
0
sage: list(G.edge_iterator())
[]  # the edge iterator is now empty
`
